pygame_gui/elements/ui_button.py

```diff
@@ -335,53 +348,59 @@ def process_event(self, event: pygame.event.Event) -> bool:
+                self.on_event(UI_BUTTON_PRESSED, {'mouse_button':event.button})
 
             if self.is_enabled and self.held:
                 self.held = False
                 self._set_inactive()
                 consumed_event = True
 
         return consumed_event
+    
+    def bind(self, event:int, function:Callable = None):
+        """
+        Bind a function to an event.
 
-    def on_start_press(self, button: int):
-        # old event to remove in 0.8.0
-        event_data = {'user_type': OldType(UI_BUTTON_START_PRESS),
-                        'ui_element': self,
-                        'ui_object_id': self.most_specific_combined_id}
-        pygame.event.post(pygame.event.Event(pygame.USEREVENT, event_data))
+        :param event: The event to bind.
 
-        # new event
-        event_data = {'ui_element': self,
-                        'ui_object_id': self.most_specific_combined_id,
-                        'mouse_button': button}
-        pygame.event.post(pygame.event.Event(UI_BUTTON_START_PRESS, event_data))
-
-    def on_pressed(self, button: int):
-        # old event
-        event_data = {'user_type': OldType(UI_BUTTON_PRESSED),
-                        'ui_element': self,
-                        'ui_object_id': self.most_specific_combined_id}
-        pygame.event.post(pygame.event.Event(pygame.USEREVENT, event_data))
+        :param function: The function to bind. None to unbind.
 
-        # new event
-        event_data = {'ui_element': self,
-                        'ui_object_id': self.most_specific_combined_id,
-                        'mouse_button': button}
-        pygame.event.post(pygame.event.Event(UI_BUTTON_PRESSED, event_data))
+        """
+        if function is None:
+            self.handler.pop(event, None)
+            return
 
-    def on_double_clicked(self, button: int):
-        # old event to remove in 0.8.0
-        event_data = {'user_type': OldType(UI_BUTTON_DOUBLE_CLICKED),
-                        'ui_element': self,
-                        'ui_object_id': self.most_specific_combined_id}
-        pygame.event.post(pygame.event.Event(pygame.USEREVENT, event_data))
+        if callable(function):
+            self.handler[event] = function
+        else:
+            raise TypeError("Function to bind must be callable")
+    
+    def on_event(self, event:int, data:Dict[str, Any]=None):
+        """
+        Called when an event occurs. Handle events.
 
-        # new event
-        event_data = {'ui_element': self,
-                        'ui_object_id': self.most_specific_combined_id,
-                        'mouse_button': button}
-        pygame.event.post(pygame.event.Event(UI_BUTTON_DOUBLE_CLICKED, event_data))
+        :param event: The event occurs.
+
+        :param data: event data
+
+        """
+        if data is None:
+            data = {}
+            
+        if event in self.handler:
+            self.handler[event](data)
```

Comment:
```suggestion
            num_params = len(signature(self.handler[event]).parameters)
            if num_params == 1:
                self.handler[event](data)
            elif num_params == 0:
                self.handler[event]()
            else:
                raise RuntimeError("Button command function signatures can have 0 or 1 parameter. "
                                   "If one parameter is set it will contain data for the id of the mouse button used "
                                   "to trigger this click event.")
```

Reply:
Will also need to import signature from inspect at the top of the file:
```python
from inspect import signature
```